### PR TITLE
Expand kernel patterns

### PR DESCRIPTION
**Pull request by rbalint**

### CODE REVIEW

File: test/autopkgtest_kernel_patterns.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import logging
+import re
+import subprocess
+import unittest
+
+from unattended_upgrade import (running_kernel_pkgs_regexp,
+                                versioned_kernel_pkgs_regexp)
+
+
+class TestSubstitude(unittest.TestCase):
```
Comment (julian-klode) [inline test/autopkgtest_kernel_patterns.py]: "Substitude" -> "Substitut"e

Comment (rbalint) [reply test/autopkgtest_kernel_patterns.py]: Fixed.

**Comment by rbalint** [comment]
@julian-klode I think it may even be better to push the regexp generation logic to pyhon-apt because update-manager needs to do the same, maybe u-r-u, too, to not remove running kernel.

**Comment by julian-klode** [comment]
I agree!

**Comment by rbalint** [comment]
@julian-klode  If LP: #1615381 could be  fixed in libapt first to protect the running kernel then only an API to list the versioned kernel-related autoremovable packages would be needed in python-apt. Then API clients could save the filtering to protect the running kernel.

**Comment by rbalint** [comment]
@julian-klode How about merging this for now to u-u, then dropping it when python-apt exposes API and APT is fixed to do parts of this?

**Pull request merged**